src/entities/Proposal/routes.ts

```diff
@@ -409,11 +410,44 @@ export async function getProposal(req: Request<{ proposal: string }>) {
 }
 
 const updateProposalStatusValidator = schema.compile(updateProposalStatusScheme)
+
+export async function commentProposalUpdateInDiscourse(id: string, caller: string) {
+  const updatedProposal: ProposalAttributes | undefined = await ProposalModel.findOne<ProposalAttributes>({ id })
+
+  if (!!updatedProposal) {
```

Comment:
Consider checking the precondition and return/throw an error right away. Then continue with the logic block.

```
if(!updatedProposal) return console.log(....);
```